### PR TITLE
Changed from read file to read bytes

### PR DESCRIPTION
**Pull request by NotArme**
Forked repository was reading a filepath to correct it's orientation, this PR attempts to change that so the image byte data can be passed directly without having to read and save files

### CODE REVIEW

File: utils.go [hunk 1]
```diff
@@ -97,14 +97,6 @@ func reverseOrientation(img image.Image, o string) *image.NRGBA {
 	case "8":
 		return imaging.Rotate90(img)
 	}
-	logrus.Errorf("unknown orientation %s, expect 1-8", o)
+	logrus.Errorf("unknown orientation %s, expect 2-8", o)
```
Comment (igorrs) [inline utils.go:100]: It seems we didn't log any errors before when orientation was 1. Shouldn't we keep it like that?
```suggestion
	logrus.Errorf("unknown orientation %s, expect 1-8", o)
```

**Review by gustavoarmelin — APPROVED** [review]

File: utils.go [hunk 2]
```diff
@@ -1,87 +1,87 @@
...
+				if err != nil {
+					fmt.Printf("error while encoding corrected image: %s", err)
+				}
+				imgBody = buffer.Bytes()
+				return &imgBody
+			case "gif":
+				buffer := new(bytes.Buffer)
+				err := gif.Encode(buffer, img, nil)
+				if err != nil {
+					fmt.Printf("error while encoding corrected image: %s", err)
```
Comment (igorrs) [inline utils.go:60]: I know it was originally like that, but it's not good practice for a library to print to standard output.
We could receive the logger as a parameter. :)

**Review by igorrs — COMMENTED** [review]
Good job. :)
Just some 💅 suggestions.

File: utils.go (hunk 2 above)
Comment (igorrs) [reply utils.go:60]: Like this:
```go
func ReadImage(imgBody []byte, logger *logrus.Entry) {
```

Comment (igorrs) [reply utils.go:60]: Kage-bunshin's logger also uses `logrus` under the hood, so we would be able to do something like this:
```go
rotatedImage := ReadImage(imageBytes, kage.Logger.Entry)
```

File: utils.go (hunk 1 above)
Comment (NotArme) [reply utils.go:100]: Yeah, but the  `if orient.String() == "1" { return &imgBody }` part at the top is supposed to return early without much processing if orientation is 1, so this log shouldn't show up in that case, should i change it to 1-8 anyway?

File: utils.go (hunk 2 above)
Comment (NotArme) [reply utils.go:60]: Okay that makes sense, ill redo these logs asap

File: utils_test.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+func TestF3(t *testing.T) {
+	F3Img, err := getImageFromFilePath("./TestImages/F-3.jpg")
+	if err != nil {
+		fmt.Printf("error: %s", err)
+	}
+	imgToSaveBytes, orientation := ReadImage(F3Img, logrus.NewEntry(logrus.StandardLogger()))
+
+	os.WriteFile("./TestImages/f3_after_tests.jpg", imgToSaveBytes, os.ModeDevice.Perm())
+	if orientation == "none" {
```
Comment (gustavoarmelin) [inline utils_test.go:19]: How do you find removed that if and use https://pkg.go.dev/github.com/stretchr/testify/assert#NotEqual
`NotEqual(t TestingT, "none", orientation, "test F3 passed with orientation %s")`

File: utils_test.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+func TestF1(t *testing.T) {
+	F1Img, err := getImageFromFilePath("./TestImages/F-1.jpg")
+	if err != nil {
+		fmt.Printf("error: %s", err)
+	}
+	_, orientation := ReadImage(F1Img, logrus.NewEntry(logrus.StandardLogger()))
+
+	if orientation == "1" {
```
Comment (gustavoarmelin) [inline utils_test.go:34]: Follow the same example for before comments it =D

File: utils_test.go [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+		return
+	}
+	t.Errorf("Expected orientation 1, found: %s", orientation)
+}
+
+func TestFnone(t *testing.T) {
```
Comment (gustavoarmelin) [inline utils_test.go:40]: This test is the same as: `TestF3` ? both make the same `assert IF =D`

Comment (NotArme) [reply utils_test.go:40]: Not exactly, the output is the same, but this one should return earlier and uses less processing if the image already has no orientation, where as TestF3 assumes an image input with `orientation 3` and does the required steps to set it to none

I should've made a better assert but for now I'm just trying to get this to work tbh

File: utils_test.go [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+	if orientation == "none" {
+		t.Logf("test Fnone passed with orientation %s", orientation)
+		return
+	}
+	t.Errorf("Expected no orientation, found: %s", orientation)
+}
+
+func getImageFromFilePath(filePath string) ([]byte, error) {
+	f, err := os.ReadFile(filePath)
```
Comment (NotArme) [inline utils_test.go:55]: Getting the test file data this way might be messing with how Exif.decode() expects header bytes on windows, if that really is the case it should work properly on qa, needs further investigation

**Review by ryoshii — APPROVED** [review]

File: utils.go (hunk 1 above)
Comment (igorrs) [reply utils.go]: Not a big deal, then. The error message will be wrong if the function is called from another context, but right now it's fine.

**Pull request merged**